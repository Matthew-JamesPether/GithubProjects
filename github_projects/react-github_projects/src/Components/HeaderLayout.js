// Imports the necessary files
import { React } from "react";
import SearchBarLayout from "./SearchBarLayout";
import "./Styles/HeaderLayout.css";

// A function that displays the appropriate header for a user
const HeaderLayout = (props) => {
  // Returns the header container
  return (
    <header className="header-container">
      <div className="heading">{props.heading}</div>
      <SearchBarLayout setUser={props.setUser} setIsLoading={props.setIsLoading}/>
    </header>
  );
}

// Declares the HeaderLayout function as the default component from this module
export default HeaderLayout;
