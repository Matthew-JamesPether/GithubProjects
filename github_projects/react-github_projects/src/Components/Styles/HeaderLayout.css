/* Styles the Header layout */
.header-container {
  height: 120px;
  border-bottom: solid orange;
}

/* Styles the heading */
.heading {
  padding-top: 10px;
  height: 50px;
  text-align: center;
  color: orange;
  font-size: 30px;
  font-weight: bolder;
}