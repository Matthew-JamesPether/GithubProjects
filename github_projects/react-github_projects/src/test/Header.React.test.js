//Imports files
import { React, useState } from "react";
import HeaderLayout from "../Components/HeaderLayout";
import renderer from "react-test-renderer";

// Test the header component
test("renders Header correctly", () => {
  const tree = renderer
    .create(<HeaderLayout heading={"GITHUB PROJECTS"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
