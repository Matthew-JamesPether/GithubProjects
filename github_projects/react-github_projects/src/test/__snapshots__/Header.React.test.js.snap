// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
<header
  className="header-container"
>
  <div
    className="heading"
  >
    GITHUB PROJECTS
  </div>
  <div
    className="searchBar-layout-container"
  >
    <div
      className="searchbar-container"
    >
      <input
        onChange={[Function]}
        placeholder="Type to search..."
        value=""
      />
      <svg
        className="search-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001q.044.06.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1 1 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0"
        />
      </svg>
    </div>
  </div>
</header>
`;

exports[`renders correctly 1`] = `
<header
  className="header-container"
>
  <div
    className="heading"
  >
    GITHUB PROJECTS
  </div>
  <div
    className="searchBar-layout-container"
  >
    <div
      className="searchbar-container"
    >
      <input
        onChange={[Function]}
        placeholder="Type to search..."
        value=""
      />
      <svg
        className="search-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001q.044.06.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1 1 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0"
        />
      </svg>
    </div>
  </div>
</header>
`;
